v4/integration/fcm/src/main/java/exchange/dydx/trading/integration/fcm/PushPermissionRequesterProtocol.kt

```diff
@@ -56,30 +60,29 @@ class PushPermissionRequester @Inject constructor(
         val localActivity = activity ?: return
         // This is only necessary for API level >= 33 (TIRAMISU)
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
-            if (ContextCompat.checkSelfPermission(localActivity, Manifest.permission.POST_NOTIFICATIONS) ==
-                PackageManager.PERMISSION_GRANTED
-            ) {
+            val permissionStatus = ContextCompat.checkSelfPermission(localActivity, Manifest.permission.POST_NOTIFICATIONS)
+            if (permissionStatus == PackageManager.PERMISSION_GRANTED) {
                 // Permission granted already. Do nothing
                 return
-            } else if (shouldShowRequestPermissionRationale(localActivity, Manifest.permission.POST_NOTIFICATIONS) && sharedPreferencesStore.read(PRIMER_SHOWN_KEY) != "true") {
+            } else if (sharedPreferencesStore.read(PRIMER_SHOWN_KEY) != "true") {
                 // Show primer if needed.
-                platformDialog.showMessage(
-                    title = abacusLocalizerImp.localize("APP.PUSH_NOTIFICATIONS_PRIMER_TITLE"),
-                    message = abacusLocalizerImp.localize("APP.PUSH_NOTIFICATIONS_PRIMER_MESSAGE"),
-                    confirmTitle = abacusLocalizerImp.localize("APP.GENERAL.OK"),
-                    cancelTitle = abacusLocalizerImp.localize("APP.GENERAL.NOT_NOW"),
-                    confirmAction = ::request,
-                )
+
+                // this currently doesn't work, leaving here to figure out later.
```

Comment:
Does it work when calling from the view model?

Reply:
this ultimately gets called from the viewmodel anyways.

i think it's a layout issue with `AlertDialog` and `BottomSheetScaffold`